tiled/client/xarray.py

```diff
@@ -369,16 +370,42 @@ def register(self, name, data_array):
 
     def dataframe(self):
         if self._dataframe is None:
-            content = self.get(
-                self.link,
-                params={
-                    "format": APACHE_ARROW_FILE_MIME_TYPE,
-                    "variable": self.variables,
-                },
-            )
-            self._dataframe = deserialize_arrow(content)
+            # If self.variables contains many and/or lengthy names,
+            # we can bump into the URI size limit commonly imposed by
+            # HTTP stacks (e.g. nginx). The HTTP spec does not define a limit,
+            # but a common setting is 4K or 8K (for all the headers together).
+            # As another reference point, Internet Explorer imposes a
+            # 2048-character limit on URLs.
+            variables = []
+            dataframes = []
+            budget = INIT_BUDGET = 2000  # number of characters
+            budget -= len(self.link)
+            CONSTANT = len("&variable=")
```

Comment:
this could be moved outside the method to not get redefined each call for a micro-small performance improvement :)

Reply:
Actually, what if you made this injectable somehow so that a unit test could add a smaller value, and a future PR could make it configurable, if desired?

Reply:
Done in 3551d0b